taui/src/components/with-authenticator.js

```diff
@@ -46,17 +49,38 @@ export default function withAuthenticator (Comp, includeGreetings = false,
     }
 
     handleAuthStateChange (state, data) {
-      this.setState({ authState: state, authData: data })
-      // Unset user profile on logout
+      const { userProfile } = this.props.data
+      // Create new emtpy profile to use when browsing anonymously
+      if (state === 'useAnonymous') {
+        const profile: AccountProfile = {
+          destinations: [],
+          hasVehicle: false,
+          headOfHousehold: ANONYMOUS_USERNAME,
+          key: ANONYMOUS_USERNAME,
+          rooms: 0,
+          voucherNumber: ANONYMOUS_USERNAME
+        }
+        this.changeUserProfile(profile)
+        // Tell auth library that the anonymous user is signed in
+        this.setState({authState: 'signedIn', authData: {username: ANONYMOUS_USERNAME}})
+      } else if (state === 'signIn' && userProfile && userProfile.key === ANONYMOUS_USERNAME) {
+        // Handle full page reload when browsing site anonymously
+        this.setState({authState: 'signedIn', authData: {username: ANONYMOUS_USERNAME}})
+      } else {
+        this.setState({authState: state, authData: data})
+      }
+
+      // Clear all local storage on logout
       if (state === 'signedOut') {
-        this.changeUserProfile(null)
+        clearLocalStorage()
```

Comment:
I think that something may be going wrong with this. If I sign in, select a voucher profile, and sign out from the map page, then if I sign in again, I am sent to the map page immediately rather than the voucher search page, and the header bar displays the profile that I selected previously.

Something similar happens when switching between anonymous and signed-in accounts: if I set up a profile anonymously and then click "Sign In" from the map page, when I sign in, I'm directed to the map page instead of the voucher search page. The header bar looks like I'm signed in, but the voucher number is blank: 
<img width="600" alt="Screen Shot 2019-03-12 at 9 23 19 AM" src="https://user-images.githubusercontent.com/447977/54203394-99f9f300-44a8-11e9-81cf-4b2450b533c2.png"> 

At that point, if I click "Edit", I receive a blank page and a console error:
`TypeError: profile.destinations is undefined`
<img width="836" alt="Screen Shot 2019-03-12 at 9 26 27 AM" src="https://user-images.githubusercontent.com/447977/54203607-f1985e80-44a8-11e9-9f36-f86f1964e669.png">

Reply:
This should be fixed now. I think the issue was setting the profile to an empty object instead of null in `signIn` in the header.